index.d.ts

```diff
@@ -7,12 +7,26 @@ type TestComponent = {
   mount(): Promise<void>;
 };
 
-export = kompot;
+type Provider<P> = {
+  component: React.ComponentType<P>;
+  props: P;
+  children: React.ComponentType<any>[];
+}
+
+export = kompot; 
 export as namespace kompot;
 
 declare namespace kompot {
-  function spy(id: string, getReturnValue?: (args: any[]) => any, stringifyArgs?: (args: any[]) => string[]): any
+  function spy(id: string, getReturnValue?: (args: any[]) => any, stringifyArgs?: (args: any[]) => string[]): () => any;
+  function spyOn(obj: object, methodName: string, spyId: string, stringifyArgs?: (args: any[]) => string[]): void;
   function kompotRequire(pathToComponent: string): { [key: string]: TestComponent };
   function init(): void;
```

Comment:
there is a difference between the global kompot and the exported functions from the kompot lib. for example, the init function can only be imported and it's not on the global, and the spyOn function is only on the global.

Reply:
I see, we might need to ditch the whole `namespace` thing then, let me see